### PR TITLE
Issue352

### LINKED ISSUE #352: Reading a file written with uproot: TTreeCache::{AddBranch,DropBranch}: unknown branch -> *
I'm taking a first stab at TTree writing and I'm running into an issue when trying to read the file back into ROOT (reading it back in uproot is working fine). ROOT is telling me
```
Error in <TTreeCache::AddBranch>: unknown branch -> *
Error in <TTreeCache::DropBranch>: unknown branch -> *
```
when I try to call `TTree::Show` on it. It seems like some kind of nameless ghost of a branch is appearing in the tree some how. I've also tested reading the uproot-originating tree via `RDataFrame` (to create ROOT histograms with `RDataFrame::Histo1D`). Things just crash.

My file is created like so:

```python
## `out_dict` is a dictionary of {name(type=str) -> data(type=numpy.ndarray)}.
## `tree` is just a string for the tree name in the file.

branches = {}
for colname, arr in out_dict.items():
    if arr.dtype == np.uint32: ## convert to int32 for now
        out_dict[colname] = out_dict[colname].astype(np.int32)
    branches[colname] = uproot.newbranch(out_dict[colname].dtype)
    log.info(f"Saving branch {colname} with dtype {out_dict[colname].dtype}")

with uproot.recreate(outfile) as f:
    f[tree] = uproot.newtree(branches, flushsize="10 MB")
    f[tree].extend(out_dict, flush=False)
```

I've tried tinkering with the flush (not declaring a `flushsize` on the tree and giving extend `flush=True`) and compression settings just to see if it might impact the error I'm seeing but it doesn't (just file size is different, as I guess is expected). [Link to file](http://webhome.phy.duke.edu/~ddavis/public/uproot352.root)

### PR DESCRIPTION
**Pull request by reikdas**
Fixes #352 

Since release 3.10.3 hasn't been released yet, we can release this alongside the other changes.

### CODE REVIEW

File: uproot/write/objects/util.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def __init__(self):
 
     def _putclass(self, cursor, obj, keycursor, beg):
         start = cursor.index - keycursor.index
-        beg = beg - keycursor.index - 8
+        beg = beg - keycursor.index
```
Comment (jpivarski) [inline uproot/write/objects/util.py]: This is the key thing. I'm just glad this makes so much sense! (I could imagine a situation in which we twiddle something, it all works again, and we just hold our breaths. Really understanding where this came from is a huge benefit.)

**Comment by reikdas** [comment]
> Do you know if TTree::Show works in ROOT? Or the TBrowser? (Start a ROOT session and type new TBrowser(); on the commandline, then double-click on the file, the tree, and the branch to see a plot. If that all works, it's good.)

It works :)

**Comment by jpivarski** [comment]
Okay, once the tests pass, merge and draft a new uproot release. Congratulations!

**Comment by reikdas** [comment]
I have drafted the release.

**Comment by jpivarski** [comment]
But let's wait until the multiple-branch problem is solved before deploying a release. Thanks!

**Pull request merged**